src/inc/til/rect.h

```diff
@@ -12,6 +12,58 @@
 
 namespace til // Terminal Implementation Library. Also: "Today I Learned"
 {
+    struct inclusive_rect

```

Comment:
This struct is the new replacement for `SMALL_RECT`. It uses the same underlying `til::CoordType` as `til::rect`, but represents a rectangle with inclusive coordinates.
The benefit is a simplified conversion between `inclusive_rect` and `rect` without overflow checking.

Reply:
I was asked whether this struct is still part of the PR title (introducing `VTInt`). I'd argue: Yes.
The goal of `VTInt` is to turn `size_t -> int` narrowings into primitive `int32_t -> int32_t` copies/casts. This requires us to replace any use of shorts in this code though, which includes any use of the `SMALL_RECT` struct. This is where `inclusive_rect` enters the scene.

Reply:
This concerns me a little, because I know there are parts of conhost that definitely use `SMALL_RECT` with exclusive coordinates. It's tricky enough as it is not knowing whether a `SMALL_RECT` is inclusive or exclusive, but it'll be a whole lot worse if we're storing exclusive coordinates in a type named `inclusive_rect`.

Or are you dealing with this on a case by case basis, and only using `til::rect` for the exclusive cases? Sorry if this is dumb question -  I've only really skimmed the PR.

Reply:
I've painstakingly replaced all inclusive `SMALL_RECT`s in our entire code base with `til::rect` as part of my `til::rect`-refactorization. This is one of the reasons I removed implicit constructors and conversion operators, as it helped me tremendously when detecting these misuses of `SMALL_RECT` and prevent any (well... most?) new bugs within my refactorization.

Basically...
> Or are you dealing with this on a case by case basis, and only using til::rect for the exclusive cases?

-> Yes! Just like in this PR, I'll use this `inclusive_rect` only for actually inclusive coordinates in follow-up PR as well. 😊